src/eddymotion/estimator.py

```diff
@@ -75,16 +77,69 @@ def fit(
             Number of parallel jobs.
         seed : :obj:`int` or :obj:`bool`
             Seed the random number generator (necessary when we want deterministic
-            estimation). If an integer, the value is used to initialize the
-            generator; if ``True``, the arbitrary value of ``20210324`` is used
-            to initialize it.
+            estimation).
+        gradient_ordering : :obj:`str`, optional
+            Specifies the ordering of the gradients. Default is "random".
+        testing : :obj:`bool`, optional
+            Flag to indicate if testing mode is enabled. Default is False.
+
         Return
         ------
         affines : :obj:`list` of :obj:`numpy.ndarray`
             A list of :math:`4 \times 4` affine matrices encoding the estimated
             parameters of the deformations caused by head-motion and eddy-currents.
+        index_order : :obj:`numpy.ndarray`
+            The shuffled index order of the gradients. Only returned if `testing` is True.
 
         """
+
+        def handle_gradient_ordering():
```

Comment:
Re: https://github.com/nipreps/eddymotion/pull/149#discussion_r1549864483

All the methods here do not access any instance variable, and any IDE will suggest them becoming static, as they were prior to 8153649. However the `actions` dictionary was triggering errors because it was taking static methods. All this could be avoided if these methods were refactored into a `utils.py` (or similar) module that takes care of sorting the DWI volumes. In fact, (i) the estimator should not take care of this at all: it should receive the DWI volumes already sorted, and (ii) these methods may be useful elsewhere.

Related to this, the strategies could be put into an `enum` so that it becomes clear which are the sorting possibilities, and they are documented and available at a single place. However, I know that using `enum`s is not very popular across many major scientific Python packages.

Reply:
Sounds like a great idea! I agree, I'll externalize the functions into a new module, util.py. Thanks for the suggestion!